File: LeetCode/209长度最小的子数组.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int start = 0, sum = 0, ans = Integer.MAX_VALUE;
        for (int i = 0; i < nums.length; i++) {
            sum += nums[i];
            while (sum >= target) {
                int length = i - start + 1;
                ans = Math.min(ans, length);
                sum -= nums[start++];
            }
        }
        return ans == Integer.MAX_VALUE ? 0 : ans;
    }
}

File: LeetCode/209长度最小的子数组.go
func minSubArrayLen(target int, nums []int) int {
	int_max := 2 << 31
	ans := int_max
	start, sum, length := 0, 0, 0
	for i := 0; i < len(nums); i++ {
		sum += nums[i]
		for sum >= target {
			length = i - start + 1
			if ans >= length {
				ans = length
			}
			sum -= nums[start]
			start++
		}
	}
	if ans == int_max {
		return 0
	}
	return ans
}